index.js

```diff
@@ -349,10 +349,12 @@ class Raspi extends EventEmitter {
 
   digitalWrite(pin, value) {
     const pinInstance = this[getPinInstance](this.normalize(pin));
-    if (pinInstance.mode != OUTPUT_MODE) {
+    if (pinInstance.mode === INPUT_MODE && value === HIGH) {
+      pinInstance.peripheral = new DigitalInput({ pin: this.normalize(pin), pullResistor: PULL_DOWN });
+    } else if (pinInstance.mode != OUTPUT_MODE) {
       this.pinMode(pin, OUTPUT_MODE);
     }
-    if (value != pinInstance.previousWrittenValue) {
+    if (pinInstance.mode === OUTPUT_MODE && value != pinInstance.previousWrittenValue) {
```

Comment:
Can you explain why you added the extra conditional here? The way it's written, it's //supposed// to change the pin mode on line 355 above so that the mode will always be OUTPUT_MODE at this point.